c++/include/orc/sargs/Literal.hh

```diff
@@ -36,6 +36,33 @@ namespace orc {
    */
   class Literal {
   public:
+    struct Timestamp {
+      Timestamp() = default;
+      Timestamp(const Timestamp&) = default;
+      Timestamp(Timestamp&&) = default;
+      ~Timestamp() = default;
+      Timestamp(int64_t second_, int32_t nano_): second(second_), nano(nano_) {
+        // PASS
+      }
+      Timestamp& operator=(const Timestamp&) = default;
+      Timestamp& operator=(Timestamp&&) = default;
+      bool operator==(const Timestamp& r) const {
+        return second == r.second && nano == r.nano;
+      }
+      bool operator<(const Timestamp& r) const {
+        return second < r.second || (second == r.second && nano < r.nano);
+      }
+      bool operator<=(const Timestamp& r) const {
+        return second < r.second || (second == r.second && nano <= r.nano);
+      }
+      bool operator!=(const Timestamp& r) const { return !(*this == r); }
+      bool operator>(const Timestamp& r) const { return r < *this; }
+      bool operator>=(const Timestamp& r) const { return r <= *this; }
+      int64_t getMillis() const { return second * 1000 + nano / 1000000; }
```

Comment:
Is `nano / 1000000` safe? It looks like `0` always due to the integer division.

Reply:
Could you add a test coverage fr `getMillis` function? I cannot find one so far.

Reply:
Added a test for getMillis. nano / 1000000 is fine because nano only has 9 digits and it will not be zero if it has more than 6 digits.